packages/types/src/codec/Struct.ts

```diff
@@ -111,6 +111,13 @@ export default class Struct<
     return class extends Struct<S> {
       constructor (value?: any, jsonMap?: Map<keyof S, string>) {
         super(Types, value, jsonMap);
+
+        (Object.keys(Types) as Array<keyof S>).forEach((key, index) => {
+          Object.defineProperty(this, key, {
```

Comment:
Does this by any chance add TS typings on the struct?

Reply:
Not 100% sure what you mean - if it is available, nope, look at the test, had to any to extract it :(